src/s1reader/s1_reader.py

```diff
@@ -682,17 +766,42 @@ def burst_from_xml(annotation_path: str, orbit_path: str, tiff_path: str,
     # calculate the range at mid swath (mid of SM swath, mid of IW2 or mid of EW3)
     with open_method(iw2_annotation_path, 'r') as iw2_f:
         iw2_tree = ET.parse(iw2_f)
-        iw2_slant_range_time = float(iw2_tree.find('imageAnnotation/imageInformation/slantRangeTime').text)
+        iw2_slant_range_time =\
+            float(iw2_tree.find('imageAnnotation/imageInformation/slantRangeTime').text)
         iw2_n_samples = int(iw2_tree.find('swathTiming/samplesPerBurst').text)
         iw2_starting_range = iw2_slant_range_time * isce3.core.speed_of_light / 2
         iw2_mid_range = iw2_starting_range + 0.5 * iw2_n_samples * range_pxl_spacing
 
     # find orbit state vectors in 'Data_Block/List_of_OSVs'
     if orbit_path:
         orbit_tree = ET.parse(orbit_path)
-        osv_list = orbit_tree.find('Data_Block/List_of_OSVs')
+        orbit_state_vector_list = orbit_tree.find('Data_Block/List_of_OSVs')
+
+        # Calculate ascending node crossing time from orbit;
+        # compare with the info from annotation
+        ascending_node_time_orbit = get_ascending_node_time_orbit(
+            orbit_state_vector_list, first_line_utc_time)
+        if ascending_node_time_orbit is None:
+            warnings.warn('Cannot estimate ascending node crossing time from Orbit. '
+                          'Using the ascending node time in annotation.')
+            ascending_node_time = ascending_node_time_annotation
+        else:
+            # Calculate the difference in the times.
+            # Give warning message when the difference is noticeable.
+            diff_ascending_node_time_seconds = (
+                ascending_node_time_orbit - ascending_node_time_annotation).total_seconds()
+            if abs(diff_ascending_node_time_seconds) > ASCENDING_NODE_TIME_TOLERANCE_IN_SEC:
+                warnings.warn('ascending node time error larger than '
+                            f'{ASCENDING_NODE_TIME_TOLERANCE_IN_SEC} seconds was detected: '
+                            f'Error = {diff_ascending_node_time_seconds} seconds.')
+
+            ascending_node_time = ascending_node_time_orbit
```

Comment:
This assignment `ascending_node_time = ascending_node_time_orbit` is outside the if statement `if abs(diff_ascending_node_time_seconds) > ASCENDING_NODE_TIME_TOLERANCE`, so it will happen for cases where the ANX time difference is less than or equal `ASCENDING_NODE_TIME_TOLERANCE`. 

Is that what you intended? I thought you'd only use `ascending_node_time_orbit` if the ANX time difference was greater than the tolerance.

Reply:
For the record:
- By default, the ANX time will be estimated from the orbit
- ANX time from the annotation will be used if:
  - The orbit file was not provided into the reader, or
  - ANX time estimation from the orbit is not successful.

The snippet above is to let the user informed that there are significant difference between the two ANX times. I thought it would be useful for the users about what is happening with the data.